crates/p2p_protocol/protocol/p2p.capnp

```diff
@@ -0,0 +1,179 @@
+# lib3h p2p wire protocol
+# note, initial handshake will use transit-enc.capnp
+# once that is complete, messages from this schema
+# will be encoded as negotiated.
+
+@0x859198991b95d3e1;
+
+struct Message {
+  # Main P2P Message struct, anon union determines message type
+
+  union {
+    msgError @0 :MsgError;
+    # indicates an error
+    # if the error is not recoverable, the connection will be closed
+
+    msgHandshake @1 :MsgHandshake;
+    # On a new connection, tell the remote node about ourselves.
+
+    msgGspArcRequest @2 :MsgGspArc;
+    # Open a gossip sequence with a remote node.
+
+    msgGspArcResponse @3 :MsgGspArc;
+    # Second stage of a gossip sequence with a remote node.
+
+    msgGspAspectDataRequest @4 :MsgGspAspectDataRequest;
+    # If we have determined a list of aspect-hashes we require
+    # from a remote node, request them.
+
+    msgGspAspectDataResponse @5 :MsgGspAspectDataResponse;
+    # A remote node has requested aspect data from us, give it to them.
+
+    msgGspAspectBroadcast @6 :MsgGspAspectBroadcast;
+    # Fast push new dht data
+
+    msgDirectRequest @7 :MsgDirect;
+    # node-to-node message request
+
+    msgDirectResponse @8 :MsgDirect;
+    # node-to-node message response
+
+    msgQueryRequest @9 :MsgQuery;
+    # dht query request
+
+    msgQueryResponse @10 :MsgQuery;
+    # dht query response
+  }
+
+  # -- top-level Message Types -- #
+
+  struct MsgError {
+    errorCode @0 :ErrorCode;
+    # code indicating if error is well-known
+
+    errorText @1 :Text;
+    # text indicating details of error
+
+    enum ErrorCode {
+      unknown @0;
+      # default if error is not well-known, or if remote is using a newer proto
+    }
+  }
+
+  struct MsgHandshake {
+    fromStoreArc @0 :UInt32;
```

Comment:
I don't get what the `from` is suppose to mean here.
Is this value the starting loc of the Arc or the Arc length?
If its the starting loc, I'd suggest renaming to `storeArcStart`
If its the Arc length I'd suggest renaming to `storeArcLength`
Also, is the length the full breadth of the arc or the half length since its suppose to be symmetric from the agent loc?

Reply:
The "from" is supposed to indicate it's the arc of the node sending the message. Any ideas for something that would be more clear?

Yep, it's the half length. `0` is no arc, `1` means covering exactly the location that their agentId hashes to (let's say 42), `2` would indicate anything (inclusively) between 41 and 43.

`sourceAgentStoreArcHalfLength`?

Reply:
I don't see how it can be the arc of someone else than the node sending the message, so I don't find this info relevant. Maybe it would if we had `fromStoreArc` and `toStoreArc` in that struct, but thats not the case so I'd go with `storeArcHalfLength` or `storeRadius` ;)

Reply:
What do you think about `storeArcRadius` and `queryArcRadius` just to be a little extra explicit?

Reply:
I think its nice 👍